clients/drcachesim/scheduler/scheduler_impl.h

```diff
@@ -758,7 +758,10 @@ template <typename RecordType, typename ReaderType> class scheduler_impl_tmpl_t
     pick_next_input_for_mode(output_ordinal_t output, uint64_t blocked_time,
                              input_ordinal_t prev_index, input_ordinal_t &index) = 0;
 
-    // Helper for pick_next_input() specialized by mapping_t mode.
+    // Helper for pick_next_input() specialized by mapping_t mode: determines
```

Comment:
Also add when the derived class will be allowed to determine whether to check for input switch.

Reply:
Oh sorry that new comment is on pick_next_input_for_mode().  What we want here is that this is called on every record, inside next_record().  Added that (and s/pick_next_input/next_record/).

Reply:
Further added to link the other way:
  
If this returns true for "need_new_input", pick_next_input_for_mode() is called.

Reply:
Updated the PR title and description as well.